### DISCUSSION TITLE
Capturing from URL-encoded body

### DISCUSSION THREAD

**Discussion by raine** [Orange-OpenSource/hurl#1707] [Q&A] [answered]
Given this hurl file

````hurl
POST https://github.com/login/device/code
Content-Type: application/x-www-form-urlencoded
```
client_id=Iv1.1bbd5e03617adebb&scope=repo+read%3Aorg
```
````

that returns

```
device_code=123&expires_in=899&interval=5&user_code=CC3B-9D56&verification_uri=https%3A%2F%2Fgithub.com%2Flogin%2Fdevice%
```

How would I parse for example the value for `device_code`?

Is using regex like so the only way?

```
[Captures]
device_code: regex "device_code=(.*?)&"
```

It seems like a not that uncommon use-case so I'd imagine there would be capturing mechanism for URL-encoded responses.

Thanks.

edit: I learned that Github can also return the response in JSON if `Accept: application/json` is given as header, so that helps.

**Comment by jcamiel** [comment, marked as answer]
Hi @raine indeed in your case the best way to extract a parameter is with a regex. There are also [filters](https://hurl.dev/docs/filters.html) to refine captures (like `split`, `urlDecode` etc...) but in this case regex is the way to go.
